### PR TITLE
Fixed some minors bugs

### PR DESCRIPTION
**Pull request by pedroaugustobt**

### CODE REVIEW

File: module/scripts/combat/defenses.js [hunk 1]
```diff
@@ -135,7 +135,7 @@ function ExecuteDefense(actor, messageId, totalAttack) {
 
 async function defense(
     actor,
-    { skillName, modifier = 0, stagger = false, block = false },
+    { skillName, modifier = 0, stagger = false, block = false, parry = false },
```
Comment (Stexinator) [inline module/scripts/combat/defenses.js:138]: This is unnecessary. See the existing parry block

File: module/scripts/combat/defenses.js [hunk 2]
```diff
@@ -94,7 +94,7 @@ function ExecuteDefense(actor, messageId, totalAttack) {
                     let defenseSkill = selectedOption.value;
                     defense(
                         actor,
-                        { skillName: defenseSkill, modifier: -3, stagger: true },
+                        { skillName: defenseSkill, modifier: -3, stagger: true, parry: true },
```
Comment (Stexinator) [inline module/scripts/combat/defenses.js:97]: This is unnecessary. See the existing parry block

File: module/scripts/combat/defenses.js [hunk 3]
```diff
@@ -421,6 +422,21 @@ function handleDefenseResults(
         if (actor.statuses.find(status => status == 'stun')) {
             actor.toggleStatusEffect('stun');
         }
+
+        if (parry) {
```
Comment (Stexinator) [inline module/scripts/combat/defenses.js:426]: You can add this to the existing parry block

File: module/setup/deprecations.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ async function globalModifiers() {
 async function specialModifiers() {
     let affectedActors = game.actors
         .filter(actor => actor.isOwner)
-        .filter(actor => actor.system.specialSkillModifiers.length > 0);
+        .filter(actor => actor.system.specialSkillModifiers && actor.system.specialSkillModifiers.length > 0);
```
Comment (Stexinator) [inline module/setup/deprecations.js:30]: An empty array is a falsy value. So this should not be necessary

File: templates/partials/character/tab-inventory.hbs [hunk 5]
```diff
@@ -118,15 +118,20 @@
...
-                                        <td></td>
-                                        <td></td>
-                                    {{/if}}
                                     <td>
                                         <div class="flex">
                                             <label class="label-info">
                                                 <b>{{localize "WITCHER.Weapon.Short.Concealment"}}: </b>
-                                            </label>{{weapon.system.conceal}}
+                                            </label>
+                                            {{#if (eq weapon.system.conceal "T")}}
```
Comment (Stexinator) [inline templates/partials/character/tab-inventory.hbs:126]: You should use lookups instead. Look at how I changed the hands for weapons

**Comment by Stexinator** [comment]
Stating the absence of one thing, does not automatically confirms the presence of another thing

File: module/setup/deprecations.js (hunk 4 above)
Comment (pedroaugustobt) [reply module/setup/deprecations.js:30]: > An empty array is a falsy value. So this should not be necessary

This error appears. Is it possible to remove this error log in another way?
image

**Comment by pedroaugustobt** [comment]
> Stating the absence of one thing, does not automatically confirms the presence of another thing.
> 
> Else you would also need to remove the reliability damage from blocking, due to the text saying parrying. As far as I can see they made a major fumble when splitting block and parry and did not clean up the texts

yes, it's clearer to me now, I asked other GMs and it seems that the damage really only applies to blocking. Sorry Stex, i'll remove the commit

**Comment by pedroaugustobt** [comment]
@Stexinator I removed the parry commit and fixed the deprecation log.
If there are any more pending issues or something that is wrong that I need to change, just tell me here!

File: module/scripts/rolls/extendedRoll.js [hunk 6]
```diff
@@ -10,6 +10,7 @@ export async function extendedRoll(rollFormula, messageData, config, flags = [])
 
     //crit/fumble calculation
     if (config.showCrit && (isCrit(roll) || isFumble(roll))) {
+        await game.dice3d?.showForRoll(roll, game.user, true);
```
Comment (Stexinator) [inline module/scripts/rolls/extendedRoll.js:13]: So this roll will only show if there is a crit or fumble?

File: templates/partials/character/tab-inventory.hbs [hunk 7]
```diff
@@ -107,18 +107,17 @@
                                                     <b>{{localize "WITCHER.Weapon.Hands.title"}}: </b>
                                                 </label>
                                                 {{ localize (lookup ../config.weapon.hands weapon.system.hands) }}
-                                            </div>
+
+                                            </div> 
+
```
Comment (Stexinator) [inline templates/partials/character/tab-inventory.hbs:112]: Why did you add empty lines?

File: module/scripts/rolls/extendedRoll.js (hunk 6 above)
Comment (pedroaugustobt) [reply module/scripts/rolls/extendedRoll.js:13]: the roll will happen normally if it is not a critical or fumble.

Comment (Stexinator) [reply module/scripts/rolls/extendedRoll.js:13]: Yes, and you only show the roll with 3d dice when it is a crit or fumble. Read the line above

**Comment by pedroaugustobt** [comment]
> Why did you close it? Dont you want to have these changes in the system?

Hey Stex, I'll redo them here, I updated my repository with the latest update (v12.15.3). I'm working on it now!
